### PR TITLE
ziggurat: allow attaching a face to a test-result for future use

### PR DESCRIPTION
**Pull request by hosted-fornet**
**Problem**:

Hard to use results from previous steps.

**Solution**:

Allow attaching a face to previous step result and add that to subject for subsequent steps.

**Notes**:

This changes `test-step`s to add a `result-face=(unit @t)` field, so a field for the face will have to be added to the FE before this can be merged @0x70b1a5

### CODE REVIEW

File: lib/zig/ziggurat.hoon [hunk 1]
```diff
@@ -2202,29 +2202,33 @@
     ?-    -.test-step
         %dojo
       %-  pairs
-      :^    ['type' %s -.test-step]
+      :~  ['type' %s -.test-step]
+          ['result_face' (result-face result-face.test-step)]
```
Comment (0x70b1a5) [inline lib/zig/ziggurat.hoon:2206]: Here and below, please change `result_face` to `result-face` ... the FE logic assumes the BE input name and FE output name are identical.

**Review by 0x70b1a5 — CHANGES_REQUESTED** [review]
`L`ittle `G`rievances. `T`hanks, `M`onkey 🔍🦧

File: lib/zig/ziggurat.hoon (hunk 1 above)
Comment (hosted-fornet) [reply lib/zig/ziggurat.hoon]: Fixed in 35fc602

**Comment by hosted-fornet** [comment]
By the way -- there is another breaking change coming. Specifically, the `path` in the [`$scry-payload`](https://github.com/uqbar-dao/dev-suite/blob/master/sur/zig/ziggurat.hoon#L88-L89) is becoming a `@t`. I'm just working out some bugs in upgrading to newest engine, then will put out that change as well.

**Review by 0x70b1a5 — APPROVED** [review]

**Pull request merged**